great_expectations/metrics/metric_results.py

```diff
@@ -27,6 +33,56 @@ class MetricErrorResultValue(TypedDict):
 class MetricErrorResult(MetricResult[MetricErrorResultValue]): ...
 
 
+class ConditionValuesValueError(ValueError):
+    def __init__(self, actual_type: type) -> None:
+        super().__init__(
+            "Value must be one of: "
+            "pandas.Series, pyspark.sql.Column, or sqlalchemy.BinaryExpression. "
+            f"Got type `{actual_type}` instead."
+        )
+
+
+class ConditionValues(MetricResult[Union[pd.Series, "pyspark.sql.Column", "BinaryExpression"]]):
+    @classmethod
+    def validate_value_type(cls, value):
+        if isinstance(value, pd.Series):
+            return value
+
+        try:
+            from great_expectations.compatibility.pyspark import pyspark
+
+            if isinstance(value, pyspark.sql.Column):
+                return value
+        except (ImportError, AttributeError):
+            pass
+
+        try:
+            from great_expectations.compatibility.sqlalchemy import BinaryExpression
+
+            if isinstance(value, BinaryExpression):
+                return value
+        except (ImportError, AttributeError):
+            pass
+
+        raise ConditionValuesValueError(type(value))
+
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate_value_type
+
+    def equals(self, other: pd.Series) -> bool:
```

Comment:
Are `equals` and `compares` methods and how they only support certain operations a result of how these metrics are computed now?

Reply:
I really did this to make the type checker happy in my tests. We could just remove these equality-checking methods and ignore the type errors.

Reply:
I removed these and added an assertion to the tests to handle it